control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go

```diff
@@ -53,10 +55,6 @@ func NewParams(hcp *hyperv1.HostedControlPlane, version string, images map[strin
 }
 
 func ReconcileClusterNodeTuningOperatorMetricsService(svc *corev1.Service) error {
-	svc.Annotations = map[string]string{
-		"service.beta.openshift.io/serving-cert-secret-name": "node-tuning-operator-tls",
-	}
-
```

Comment:
removing the annotation here does not remove it from an existing service in the upgrade case

Reply:
Thanks @sjenning, good catch. I switched this to remove the `"service.beta.openshift.io/serving-cert-secret-name"` entry from the svc.Annotations Map if it exists. 

Does that seem like a reasonable solution to you?

Reply:
Hey @sjennings, I see now that this won't work because the PKI is reconciled before NTO, and the control-plane-operator can't overwrite the secret generated by the service-ca operator. 

Do you know if this problem has already been solved for any other HyperShift components that previously used service-ca certs?

Otherwise, I suppose when we reconcile the NTO certs we should check if the `service/node-tuning-operator` contains the `service.beta.openshift.io/serving-cert-secret-name` annotation, and if it does remove it and delete the existing `node-tuning-operator-tls` secret. Not sure if there is a cleaner solution that I am missing...